src/main/java/com/hillel/items_exchange/service/impl/ImageServiceImpl.java

```diff
@@ -136,4 +145,33 @@ private void validateImagesTypes(List<MultipartFile> images) throws UnsupportedM
                 .filter(Predicate.not(supportedTypes::contains))
                 .collect(Collectors.toSet());
     }
+
+    @Override
+    public byte[] scale(byte[] bytes) throws IOException {
+        try (ByteArrayInputStream bais = new ByteArrayInputStream(bytes);
+             ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+
+            BufferedImage originImage = ImageIO.read(bais);
+            Dimension newSize = calculatePreferThumbnailSize(
+                    new Dimension(originImage.getWidth(), originImage.getHeight()));
+            BufferedImage scaledImage = getScaled(newSize, originImage);
+            String type = URLConnection.guessContentTypeFromStream(new ByteArrayInputStream(bytes)).replace("image/", "");
+            ImageIO.write(scaledImage, type, baos);
+            return baos.toByteArray();
+        }
+    }
+
+    private BufferedImage getScaled(Dimension d, BufferedImage originImage) {
+        java.awt.Image scaledInstance = originImage.getScaledInstance(d.width, d.height, SCALE_SMOOTH);
```

Comment:
Why do you use java.awt.Image here instead of importing the package and to use only Image?

Reply:
Because the class already has had the import `com.hillel.items_exchange.model.Image;`. It is a name conflict.

Reply:
Ok, got it.